### ISSUE TITLE
Characters missing in the font used for Patch Browser. Using MacOS Garageband

### ISSUE THREAD

**Issue by tomhannen** [surge-synthesizer/surge#7021]
**Bug Description:**
When I click on the patch name in Surge XT, I see a series of question marks, suggesting missing Unicode symbols in the font used.

**Surge XT Version**
1.2.3.bb8dd2b

* Version: 1.2.3.bb8dd2b
* Plugin Type: AU
* Bitness: 64-bit

**Reproduction Steps:**
1. Open Surge XT as an instrument in Garageband
2. Click on 'Init Saw'
3. See error - incorrect characters displayed as question marks

**Expected Behavior:**
Font should render correctly

**Screenshots:**
![image](https://github.com/surge-synthesizer/surge/assets/185299/d525f0da-caea-4f83-b278-c82e02b1bc1a)

**Computer Information (please complete the following!):**
 - OS: MacOS 13.3.1(a)
 - Host: Garageband 
 - Version: 10.4.8

**Additional Information:**
NA

**Comment by baconpaul** [comment]
Oh that’s interesting. It means it can’t make the bold font from the ttf. Those are just bold headers that say factory presets and stuff.

Can I ask you to try a couple of things? 

1. If you run the standalone (installed in applications) do you see the same?
2. What is 13.3.1(a).
3. Is your system set up for us English or some other region? If another region which?
4. If you try some other menu (like right mouse on a param) I assume you see the same?

my theory is Ventura I stricter about promoting fonts to bold equivalents from memory somehow. But that’s just a theory

**Comment by tomhannen** [comment]
Thanks for the quick reply.  Answers below:

1.  I don't see the same issue when I run the standalone app.  I was asked for two permissions dialog boxes before the app opened (one about access to the Documents folder, and one I just clicked through)..
2. 13.3.1(a) is the version of MacOS (Ventura) I am running - it's the latest one with Apple's recent security update - hence the "a" (I think?)
3. My system is English (UK).
4. I do. Below is me right-clicking a parameter in the Init Saw patch:
![image](https://github.com/surge-synthesizer/surge/assets/185299/6d52c45d-3f5f-4845-b541-08d73c10ee4f)

**Comment by baconpaul** [comment]
huh ok I was guessing the answer to (1) would be that you do see the same. Interesting. So this is something about the AU hosting in GB. I just tried in garage band 10.4.8 here with 1.2.3 on macOS 12.6 and didn't see the corruption. 

So quickly do you have any other DAWs (reaper, logic, etc...)? Fine if not but it would help narrow it down. 
Are you ARM or Intel?

with the answers to those I'll have to look at the code to make sure we are loading the bold font face properly, basically. We might not be, and it could be that GarageBand (or the au hosting service) has a specific 'fail' font fallback which is different than the standalone. 

Thanks for getting back to me.

**Comment by tomhannen** [comment]
I'm on an M1 MacBook Pro, so Apple Silicon, not Intel

I've just installed Reaper, and opened the synth in there (never used it before).
Font is working fine:
<img width="1527" alt="image" src="https://github.com/surge-synthesizer/surge/assets/185299/3a58b2ec-d0dc-405a-bf80-d61d800acb37">

**Comment by baconpaul** [comment]
OK great so we've narrowed this down to "bold fonts in garage band on ventura dont' render".

Thank you. That's a super specific report. Let me stare at the code in the next week or so and I'll tag here if i see something which might help. 

(My current theory: We use the same typeface to life to Bold for those headers; we also load the bold ttf. but i wonder if that logic on macOS is wrong and we arne't getting a ttf; and the garage band hosting environment doesn't fall back to the underlying non-bold font properly).

**Comment by baconpaul** [comment]
Specifically

```
    if (isBoldened)
    {
        g.setFont(getLookAndFeel().getPopupMenuFont().boldened());
    }
```

that is failing in GB for TTF fonts.

**Comment by Vensdale** [comment]
Same bug here on Logic Pro v10.7.8-9 on Mac Studio\Monterey 12.6.7
Video: https://youtu.be/riROQpJ2ra8

**Comment by baconpaul** [comment]
<img width="599" alt="Screen Shot 2023-07-13 at 8 34 36 AM" src="https://github.com/surge-synthesizer/surge/assets/13387561/e9a0dc9a-f08d-4f27-bbb6-5b085e9b864f">

Hmm now here's me in LPX 10.7.9 on Monterey 12.6

I thought it might be because I have the lato font installed on my system but its not that.

I'll try and make a nightly with this bold turned off on macOS and see how that works (or with the bold font shipped and use that rather than .bolden) and will update this issue when I have.

**Comment by baconpaul** [comment]
Ahhh I think I see what's happening - stay tuned.

**Comment by baconpaul** [comment]
OK I merged a change to the nightly which I think will fix this. Basically we had the bold typeface loaded but one code path didn't access it in a way which was coherent. My guess is that fell back to default font for you and then boldened it which on your system led to the mis-render.

In about an hour or so there will be a new nightly build. You can install it https://surge-synthesizer.github.io/nightly_XT and test.

If it works, you can use the nightly but please remember that it may change and especially the new features in it may change before a production release. Also fi you use the nightly then go back to the stable version, when you load patches it will give a version mis-match warning. (If you don't use new features this won't matter though). Or you can just install the nightly and run it! Or install the nightly then install 1.2.3 back under it.

If this fixes the problem and it is more widespread we could also merge into a 1.2.4 quickly but would need confirmation of the fix first.

**Comment by Vensdale** [comment]
I've installed NIGHTLY-2023-07-13-513749c built at 2023-07-13 14:06:01 UTC over v1.2.3 and it looks like the issue is gone))

![изображение](https://github.com/surge-synthesizer/surge/assets/139464671/14ddd19a-fb60-4fa1-a96b-902baaf835b1)

Thank you so much!

**Comment by mkruselj** [comment]
Fixed by #7093

**Comment by baconpaul** [comment]
Awesome! 

If water condition makes it happen becomes more prone t maybe I’ll merge this into a 1.2.4. Gold it fixed it.
